stable/weave-scope/charts/weave-scope-frontend/templates/_helpers.tpl

```diff
@@ -0,0 +1,50 @@
+{{/* Helm standard labels */}}
+{{- define "weave-scope-frontend.helm_std_labels" }}
+chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+heritage: {{ .Release.Service }}
+release: {{ .Release.Name }}
+app: {{ template "toplevel.name" . }}
```

Comment:
I don't like `toplevel`. I think it is too unspecific. Convention is to use the chart name, which would be `weave-scope`.

Reply:
The trouble with using the conventional "chart-name.{name,fullname}" templates here is that even though the parent sets the definition all children use, they render differently in different children because .Chart.Name is based on the chart that renders the template, not the chart that defines it.  So the "app" attribute would be set differently per child chart, which is not what I want if the full Weave Scope chart is being used.  What I needed here was a reference to the parent chart name, that all children would render identically.

The idea behind the "toplevel" construct (cf. kubernetes/helm#3166) is to allow a chart to say "whatever top-level ancestor (parent, grandparent, etc.) chart is ultimately depending on us, use that chart's name here, or use the name of this chart if there is no ancestor" which is exactly what I was trying to do here (it's intentionally more generic than referring to the ancestor chart by name since a child won't necessarily know what ancestors it has).  If 3166 actually gets written and merged then there would be no use for the "toplevel" named templates as such, but in cases like this it provides a workaround where people have otherwise been asking how to export values from parent charts down to child charts (which Helm doesn't support yet as far as I know) so they can set something consistently in all children without needing to set globals.